File: 10주차-스택+큐/w10-Kaya/10828.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.reflect.Array;
import java.util.*;

public class Main {
    public static Stack<Integer> s = new Stack<>();
    public static void main(String[] args) throws IOException {
        // write your code here
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int T = Integer.parseInt(br.readLine());

        while(T>0){
            String temp = br.readLine();
            String[] parse = temp.split(" ");

            if(parse.length == 2){
                // push
                s.push(Integer.parseInt(parse[1]));
            }
            else{
                // top, size, empty, pop
                switch (parse[0]){
                    case "pop":
                        if(s.empty()){
                            sb.append("-1").append("\n");
                        }
                        else {
                            sb.append(s.peek()).append("\n");
                            s.pop();
                        }
                        break;
                    case "size":
                        sb.append(s.size()).append("\n");
                        break;
                    case "empty":
                        if(s.empty()){
                            sb.append("1").append("\n");
                        }
                        else {
                            sb.append("0").append("\n");
                        }
                        break;
                    case "top":
                        if(s.empty()){
                            sb.append("-1").append("\n");
                        }
                        else {
                            sb.append(s.peek()).append("\n");
                        }
                        break;
                    default:
                        break;
                }
            }
            T--;
        }
        System.out.println(sb);
    }
}

File: 10주차-스택+큐/w10-Yoon/10828.py
import sys

n = int(sys.std.readline())
stack=[]

for i in range(n):
    stn = sys.stdin.readline().split()
    order = stn[0]

    if order == "push":
        value = stn[1]
        stack.append(int(value))

    elif order == "pop":
        if len(stack)==0:
            print(-1)
        else:
            print(stack.pop())

    elif order == "size":
        print(len(stack))

    elif order == "empty":
        if len(stack)==0:
            print(1)
        else:
            print(0)
    
    elif order == "top":
        if len(stack)==0:
            print(-1)
        else:
            print(stack[-1])

